app/merginapi.cpp

```diff
@@ -1025,77 +1119,80 @@ ProjectList MerginApi::parseProjectsData( const QByteArray &data, bool dataFromS
       }
       p.created = QDateTime::fromString( projectMap.value( QStringLiteral( "created" ) ).toString(), Qt::ISODateWithMs ).toUTC();
       QDateTime updated = QDateTime::fromString( projectMap.value( QStringLiteral( "updated" ) ).toString(), Qt::ISODateWithMs ).toUTC();
-
-      if ( dataFromServer )
-      {
-        if ( !updated.isValid() )
-        {
-          updated = p.created;
-        }
-        p.serverUpdated = updated;
-      }
-      else
+      if ( !updated.isValid() )
       {
-        p.lastSync = QDateTime::fromString( projectMap.value( QStringLiteral( "lastSync" ) ).toString(), Qt::ISODateWithMs ).toUTC();
-        if ( !updated.isValid() )
-        {
-          updated = p.created;
-        }
-        p.updated = updated;
+        updated = p.created;
       }
+      p.serverUpdated = updated;
       result << std::make_shared<MerginProject>( p );
     }
   }
   return result;
 }
 
-bool MerginApi::cacheProjectsData( const QByteArray &data )
+std::shared_ptr<MerginProject> MerginApi::parseProjectData( const QString &projectPath, const QString &cacheName )
```

Comment:
I would rename this to readProjectMetadata() - and get rid of the cacheName argument